spring-kafka/src/main/java/org/springframework/kafka/annotation/RetryTopicRegistryPostProcessor.java

```diff
@@ -0,0 +1,112 @@
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.annotation;
+
+import java.util.Arrays;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.beans.factory.support.RootBeanDefinition;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.core.Ordered;
+import org.springframework.core.type.AnnotationMetadata;
+import org.springframework.kafka.retrytopic.RetryTopicBootstrapper;
+import org.springframework.kafka.retrytopic.RetryTopicConfiguration;
+import org.springframework.kafka.retrytopic.RetryTopicInternalBeanNames;
+
+/**
+ * A {@link BeanDefinitionRegistryPostProcessor} implementation that registers
+ * the non-blocking delayed retries feature's components' beans if a
+ * {@link RetryableTopic} annotated method or a {@link RetryTopicConfiguration}
+ * bean is found.
+ *
+ * @author Tomaz Fernandes
+ * @since 2.9.0
+ *
+ * @see RetryTopicBootstrapper
+ * @see RetryTopicConfiguration
+ * @see RetryableTopic
+ */
+public class RetryTopicRegistryPostProcessor
+		implements BeanDefinitionRegistryPostProcessor, Ordered, ApplicationContextAware {
+
+	private static final String RETRY_TOPIC_BOOTSTRAPPER_BEAN_NAME = RetryTopicInternalBeanNames
+			.RETRY_TOPIC_BOOTSTRAPPER;
+	static final String RETRY_TOPIC_BOOTSTRAPPED_ATTRIBUTE = "retryTopicBootstrapped";
+	private ApplicationContext applicationContext;
+
+	@Override
+	public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry) throws BeansException {
+		if (registry.containsBeanDefinition(RetryTopicInternalBeanNames
+				.RETRY_TOPIC_BOOTSTRAPPER) && registry.getBeanDefinition(RetryTopicInternalBeanNames
+				.RETRY_TOPIC_BOOTSTRAPPER).hasAttribute(RETRY_TOPIC_BOOTSTRAPPED_ATTRIBUTE)) {
+
+			return;
+		}
+
+		Arrays
+				.stream(registry.getBeanDefinitionNames())
+				.map(registry::getBeanDefinition)
+				.filter(this::hasRetryTopicConfiguration)
+				.findFirst()
+				.ifPresent(beanDef -> bootstrapRetryTopic(registry));
+	}
+
+	private boolean hasRetryTopicConfiguration(BeanDefinition beanDef) {
+		return RetryTopicConfiguration.class.isAssignableFrom(beanDef.getResolvableType().toClass())
+				|| AnnotationMetadata
+					.introspect(beanDef.getResolvableType().toClass())
+					.hasAnnotatedMethods(RetryableTopic.class.getName());
+	}
+
+	private void bootstrapRetryTopic(BeanDefinitionRegistry registry) {
+		BeanDefinition beanDef = getOrRegisterBeanDefinition(registry);
+		this.applicationContext.getBean(RETRY_TOPIC_BOOTSTRAPPER_BEAN_NAME, RetryTopicBootstrapper.class)
```

Comment:
we cannot call `getBean()` in the `postProcessBeanDefinitionRegistry()`. This is the same "Inversion of Control impedance" I mentioned in other place.
That's too early to deal with beans.

Reply:
I see. I think I can find a way around this, but it might require more changes than I'd like to.

Reply:
I mean changes to existing components such as the `RetryTopicBootstrapper`.